src/Storages/StorageMaterializedView.cpp

```diff
@@ -97,11 +97,17 @@ StorageMaterializedView::StorageMaterializedView(
                                                                      storage_metadata.columns,
                                                                      local_context->getGlobalContext());
 
-    if (query.sql_security)
-        storage_metadata.setSQLSecurity(query.sql_security->as<ASTSQLSecurity &>());
+    ASTPtr sql_security = query.sql_security;
+    if (!sql_security)
```

Comment:
Is this `if` still necessary? I mean what if you just leave `sql_security` nullptr here and then continue without calling `storage_metadata.setSQLSecurity()`? It seems it will work as `INVOKER` anyway in this case.